### PR TITLE
Use proper Option methods instead of .get in scaladoc ModelFactory.

### PR DESCRIPTION
**Pull request by vigdorchik**

### CODE REVIEW

**Comment by jsuereth** [comment]
Review by @VladUreche or @heathermiller .

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 1]
```diff
@@ -143,8 +143,10 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
           else None
         if (sym.isPrivate) PrivateInTemplate(inTpl)
         else if (sym.isProtected) ProtectedInTemplate(qual getOrElse inTpl)
-        else if (qual.isDefined) PrivateInTemplate(qual.get)
-        else Public()
+        else qual match {
+          case Some(q) => PrivateInTemplate(q)
+          case None => Public()
+        }
       }
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala:150]: Alternatively,

```
else qual.fold(Public(): Visibility)(PrivateInTemplate)
```

File: src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 2]
```diff
@@ -876,20 +878,12 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
...
 
-    def makeNoDocTemplate(aSym: Symbol, inTpl: TemplateImpl): NoDocTemplateImpl = {
-      val bSym = normalizeTemplate(aSym)
-      noDocTemplatesCache.get(bSym) match {
-        case Some(noDocTpl) => noDocTpl
-        case None => new NoDocTemplateImpl(bSym, inTpl)
-      }
-    }
+    def makeNoDocTemplate(aSym: Symbol, inTpl: TemplateImpl): NoDocTemplateImpl =
+      noDocTemplatesCache get aSym getOrElse new NoDocTemplateImpl(aSym, inTpl)
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala]:
```
noDocTemplatesCache.getOrElse(aSym, new NoDocTemplateImpl(aSym, inTpl))
```

**Comment by paulp** [comment]
This is clearly a big improvement (LGTM) but two more common indirections which can be defeated (as noted above) are:
- get followed by getOrElse is 2-arg getOrElse
- map followed by getOrElse is fold

**Comment by paulp** [comment]
LGTM.

**Pull request merged**